ui/app/ducks/gas/gas.duck.js

```diff
@@ -416,114 +420,111 @@ export function fetchGasEstimates(blockTime) {
+        estimatedPricesAndTimes,
+        ({ expectedTime }) => expectedTime,
+      )
+
+      const withSupplementalTimeEstimates = flatten(
+        estimatedTimeWithUniquePrices.map(
+          ({ expectedWait, gasprice }, i, arr) => {
+            const next = arr[i + 1]
+            if (!next) {
+              return [{ expectedWait, gasprice }]
+            }
+            const supplementalPrice = getRandomArbitrary(
+              gasprice,
+              next.gasprice,
+            )
+            const supplementalTime = extrapolateY({
+              higherY: next.expectedWait,
+              lowerY: expectedWait,
+              higherX: next.gasprice,
+              lowerX: gasprice,
+              xForExtrapolation: supplementalPrice,
             })
-        : Promise.resolve(
-            priceAndTimeEstimates.length
-              ? priceAndTimeEstimates
-              : loadLocalStorageData('GAS_API_ESTIMATES'),
-          )
-
-    return promiseToFetch.then((estimates) => {
-      dispatch(setPricesAndTimeEstimates(estimates))
-      dispatch(gasEstimatesLoadingFinished())
-    })
+            const supplementalPrice2 = getRandomArbitrary(
+              supplementalPrice,
+              next.gasprice,
+            )
+            const supplementalTime2 = extrapolateY({
+              higherY: next.expectedWait,
+              lowerY: supplementalTime,
+              higherX: next.gasprice,
+              lowerX: supplementalPrice,
+              xForExtrapolation: supplementalPrice2,
+            })
+            return [
+              { expectedWait, gasprice },
+              {
+                expectedWait: supplementalTime,
+                gasprice: supplementalPrice,
+              },
+              {
+                expectedWait: supplementalTime2,
+                gasprice: supplementalPrice2,
+              },
+            ]
+          },
+        ),
+      )
+      const withOutliersRemoved = inliersByIQR(
+        withSupplementalTimeEstimates.slice(0).reverse(),
+        'expectedWait',
+      ).reverse()
+      const timeMappedToSeconds = withOutliersRemoved.map(
+        ({ expectedWait, gasprice }) => {
+          const expectedTime = new BigNumber(expectedWait)
+            .times(Number(blockTime), 10)
+            .toNumber()
+          return {
+            expectedTime,
+            gasprice: new BigNumber(gasprice, 10).toNumber(),
+          }
+        },
+      )
+
+      const timeRetrieved = Date.now()
+      dispatch(setApiEstimatesLastRetrieved(timeRetrieved))
+
+      await Promise.all([
+        setStorageItem('GAS_API_ESTIMATES_LAST_RETRIEVED', timeRetrieved),
+        setStorageItem('GAS_API_ESTIMATES', timeMappedToSeconds),
+      ])
+
+      estimates = timeMappedToSeconds
+    } else if (priceAndTimeEstimates.length) {
+      estimates = priceAndTimeEstimates
+    }
+
+    dispatch(setPricesAndTimeEstimates(estimates))
+    dispatch(gasEstimatesLoadingFinished())
+    return estimates
```

Comment:
This used to return nothing beforehand, rather than `estimates`. Was this change intentional?

If this is removed, then line 429 above can be moved into a condition as well, so that `estimate` is only fetched in the case where it is used.

Reply:
Lint complains `410:37  error  Expected to return a value at the end of async arrow function`; it seems logical to always return estimates from this function

Reply:
That error is from the `consistent-return` rule; it indicates that a value is returned in some branches but not others. It's being triggered by the `return null` early on in the function, and it can be addressed by returning `undefined` instead.

I'm not sure why we'd want to introduce a return value if it's never used :thinking:

Reply:
Null